### DISCUSSION TITLE
Switch libstfl formula's repo to https://github.com/newsboat/stfl

### DISCUSSION THREAD

**Discussion by Minoru** [Homebrew/discussions#4038] [Tap maintenance and Homebrew development]
Hi! I maintain Newsboat. Homebrew recently disabled libstfl formula, because its upstream repo is gone. This broke Newsboat's CI. To add insult to injury, it also made it harder for developers and users to build Newsboat from source, because they now have to build stfl too.

Homebrew gave us a heads up about this change (https://github.com/newsboat/newsboat/issues/1881), and I mentioned that Newsboat maintains an STFL fork. I thought Homebrew will switch libstfl formula to that, but instead, libstfl was disabled and newsboat now bundles the aforementioned fork.

`git grep` shows that newsboat is the only libstfl user in Homebrew. Would it be possible to re-enable libstfl formula and make it depend on https://github.com/newsboat/stfl ? Or maybe there is some other way to make stfl installable from Homebrew again?

**Comment by Bo98** [comment]
Looks like upstream is now available here: http://bygone.clairexen.net/stfl/, so we could use that.

**Comment by Minoru** [comment]
Only tarballs are available. The SVN repo is unreachable.

Also, the upstream doesn't have two fixes that Dennis van der Schagt created while working on Newsboat. These patches fix issues that affect Newsboat users and can't be worked around in Newsboat proper:
- https://github.com/newsboat/stfl/commit/0cc16e7d2784d3946c2da6cb83c78abe8dbc975c
- https://github.com/newsboat/stfl/commit/7b9f793ba239ca14025c9892ee412b4433126d62

Both were submitted upstream via the mailing list, and got no response. (I wish I could link you to the list archives, but it's down too.)

Finally, [I emailed the author two months ago](https://github.com/newsboat/stfl/issues/2#issuecomment-1250340466), and still haven't got a response.

I'd say that the upstream is available in name only.

**Reply by Bo98** [reply]
> Only tarballs are available. The SVN repo is unreachable.

This is not a concern as we do not use this.

> Also, the upstream doesn't have two fixes that Dennis van der Schagt created while working on Newsboat. These patches fix issues that affect Newsboat users and can't be worked around in Newsboat proper

Given newsboat appears to be the only thing we package that actually used stfl, we ended up bundling the dependency within newsboat itself which containes the patches: https://github.com/Homebrew/homebrew-core/blob/f34ea3b99000688f0357dc6381c070e17ec736cd/Formula/newsboat.rb#L38-L46.

In fact it was apparently said at the time that your team does not want to be the new upstream of stfl as you planned to move away from it.

To become the new upstream of an existing formula, you would need to be committed to that role of upsteam. The bar is admittedly much lower for disabled formula, but I'm not entirely sure if the situation has changed much and stfl is still ultimately dead.

Our formula never contained those patches, but our newsboat build did. Maybe I've misunderstood the situation, but given newsboat is the only formula that needs stfl, and also needs patches to make it suitable for newsboat, it sounds to be like building a copy as a part of the newsboat build process like we currently are still seems like the best option, and one that fits the temporary nature that was intended. Or has things changed since then?
